src/compiler/scala/tools/nsc/classpath/ZipAndJarFileLookupFactory.scala

```diff
@@ -181,3 +175,27 @@ object ZipAndJarSourcePathFactory extends ZipAndJarFileLookupFactory {
 
   override protected def createForZipFile(zipFile: AbstractFile): ClassPath = ZipArchiveSourcePath(zipFile.file)
 }
+
+final class FileBasedCache[T] {
+  private case class Stamp(lastModified: FileTime, fileKey: Object)
+  private val cache = collection.mutable.Map.empty[java.nio.file.Path, (Stamp, T)]
+
+  def getOrCreate(path: java.nio.file.Path, create: () => T): T = cache.synchronized {
+    val lastModified = Files.getLastModifiedTime(path)
+    val attrs = Files.readAttributes(path, classOf[BasicFileAttributes])
+    val stamp = Stamp(lastModified, attrs.fileKey())
```

Comment:
It seems `attrs.fileKey` is null on windows, at least on the version i checked. I guess that's fine, `lastModified` should be enough? Should we check still keep the `fileKey` check for other platforms?

Reply:
Yes, on other platforms we fall back to just the timestamp check. I could declare it as `Option[Object]` to make it clearer that we've thought about this.

Reply:
It's fine without an Option, a commend will do.

Reply:
pushed a comment